testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2721,6 +2721,7 @@ public void testInsertIntoNotNullColumn()
             assertQuery("SELECT * FROM " + table.getName(), "VALUES (NULL, 2)");
             // This is enforced by the engine and not the connector
             assertQueryFails(format("INSERT INTO %s (not_null_col, nullable_col) VALUES (NULL, 3)", table.getName()), "NULL value not allowed for NOT NULL column: not_null_col");
+            assertQueryFails(format("INSERT INTO %s (not_null_col, nullable_col) VALUES (TRY(5/0), 4)", table.getName()), "NULL value not allowed for NOT NULL column: not_null_col");
```

Comment:
This test passes without the other changes, at least when run with `TestIcebergOrcConnectorTest`.
See https://github.com/trinodb/trino/issues/13434#issuecomment-1203643757 for why this can be the case.

Also, a test case should be also added to all classes overriding this method (clickhouse)